spec/rspec/mocks/partial_double_spec.rb

```diff
@@ -383,6 +383,22 @@ def defined_private_method
         object.implemented
       end
 
+      it 'wraps proxies in a reentrant mutex so we dont trigger deadlocks' do
```

Comment:
> We don't have any specs that access RSpec::Mocks.space.proxy_for in this file. This bug kinda straddles things but since we're fixing it by making the space proxy mutex re-entrant it feels to me like it belongs there.

I'm trying to write a spec that will trigger the deadlock without the callback so we can do away with this, but struggling currently.

>  If you feel strongly about it staying here let's go with it.

I don't feel that strongly about it, but the deadlock error is trigger by use of the callback